src/prompt.c

```diff
@@ -52,12 +53,16 @@ void	prompt_init_buffer(t_prompt *_prompt)
 	snprintf(_prompt->buffer + _prompt->host_index, ft_strlen(_prompt->hostname) + 1, "%s", _prompt->hostname);
 	snprintf(_prompt->buffer + _prompt->colon_index, 2, ":");
 	snprintf(_prompt->buffer + _prompt->path_index, ft_strlen(_prompt->current_path) + 1, "%s", _prompt->current_path);
-	snprintf(_prompt->buffer + _prompt->bracket_index, 2, ">");
+	snprintf(_prompt->buffer + _prompt->bracket_index, 3, "> ");
 }
 
 void	prompt_update_current_path(t_prompt *_prompt)
 {
-	_prompt->current_path = directory_get_current_dir();	
+	char *old;
+
+	old = _prompt->current_path;
+	_prompt->current_path = directory_get_current_dir();
+	free(old);
```

Comment:
might as well free it immediately, then reassign it ? doesn't matter much though.